contributing.md

```diff
@@ -0,0 +1,134 @@
+
+- IRC: `#activity-stream` on `irc.mozilla.org`
+- Mailing List: [activity-stream-dev](https://groups.google.com/a/mozilla.com/d/forum/activity-stream-dev)
+
+## Architecture ##
+
+Activity stream is a Firefox addon.  It uses the Firefox [Addon SDK](https://developer.mozilla.org/en-US/Add-ons/SDK) and
+is written in Javascript.  One of the cool things about Activity Stream is that the 
+[content side of the addon](https://developer.mozilla.org/en-US/Add-ons/SDK/Guides/Content_Scripts)
+is written using [ReactJS](https://facebook.github.io/react/).  This makes it an awesome project for React hackers to contribute to!
+
+## Finding Bugs, Filing Tickets, Earning Karma ##
+
+Activity Stream lives on [GitHub](https://github.com/mozilla/activity-stream), but you already knew that!  If you've found
+a bug, or have a feature idea that you you'd like to see in Activity Stream, follow these simple guidelines:
+- Pick a thoughtful and terse title for the issue (ie. *not* Thing Doesn't Work!)
+- Make sure to mention your Firefox version, OS and basic system parameters (eg. Firefox 49.0, Windows XP, 512KB RAM)
+- If you can reproduce the bug, give a step-by-step recipe
+- You probably don't want to tag the bug, but if you must, pick either `Bug` or `Feature`
+- Include [stack traces from the console(s)](https://developer.mozilla.org/en-US/docs/Mozilla/Debugging/Debugging_JavaScript) where appropriate
+- Screenshots welcome!
+- When in doubt, take a look at some [existing issues](https://github.com/mozilla/activity-stream/issues) and emulate 
+
+
+## Take a Ticket, Hack some Code ##
+
+If you want to write some code, you need to first take a look at the current [Milestone](#milestones-iterations-national-parks-of-canada)
+to get an idea what we are currently working on, and to determine the projects [priorities](#priorities-comments).  You can then grab a ticket, hack some code,
+open up a [Pull Request](#pull-requests), get your code [reviewed](#code-reviews), and see your code merged into 
+the Activity Stream codebase.
+
+## Milestones, Iterations, National Parks of Canada ##
+
+All work on Activity Stream is broken into two week iterations, which we map into a GitHub [Milestone](https://github.com/mozilla/activity-stream/milestones).  At the beginning of the iteration, we prioritize and estimate tickets
+into the milestone, attempting to figure out how much progress we can make during the iteration.  
+
+## Priorities, Comments ##
+
+All tickets that have been [triaged](#triage) will have a priority tag of either `P1`, `P2`, or `P3`, which are highest to lowest
+priorities of tickets in Activity Stream.  We love ticket tags and you might also see `Blocked`, `Critical` or `Chemspill` tags, which
+indicate our level of anxiety about that particular ticket.  
+
+If you are new to the repo, you might want to pay close attention to `Good first bug`, `Bug`, `Chore` and `Polish` tags, as these are
```

Comment:
We may want to link these to GitHub issues. For example:

- "Good first bug" => https://github.com/mozilla/activity-stream/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+first+bug%22
- "Bug" => https://github.com/mozilla/activity-stream/issues?q=is%3Aopen%20is%3Aissue%20label%3ABug%20
- "Chore" => https://github.com/mozilla/activity-stream/issues?q=is%3Aopen+is%3Aissue+label%3AChore
- "Polish" => https://github.com/mozilla/activity-stream/issues?q=is%3Aopen+is%3Aissue+label%3APolish